library/src/blas1/rocblas_scal_batched.hpp

```diff
@@ -0,0 +1,92 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#ifndef __SCAL_BATCHED_HPP__
+#define __SCAL_BATCHED_HPP__
+
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "scal_host.hpp"
+#include "utility.h"
+
+template <typename T, typename = T>
+constexpr char rocblas_scal_name[] = "unknown";
+template <>
+constexpr char rocblas_scal_name<float>[] = "rocblas_sscal_batched";
+template <>
+constexpr char rocblas_scal_name<double>[] = "rocblas_dscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex>[] = "rocblas_cscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex>[] = "rocblas_zscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex, float>[] = "rocblas_csscal_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex, double>[] = "rocblas_zdscal_batched";
+
+template <typename T, typename U>
+rocblas_status rocblas_scal_batched_impl(rocblas_handle handle,
```

Comment:
_impl suffix convention I see in blas2 but nowhere else in blas1.   Do we want it here?

Reply:
> _impl suffix convention I see in blas2 but nowhere else in blas1. Do we want it here?

`_impl` is used for a C++ templated implementation with logging and argument error checking and device memory allocation (if any) turned on.

`_template` is used for a C++ templated implementation without logging, argument error checking, or device memory allocation, so that `_template` can be called from other rocBLAS without incurring overhead.

`#2` in: https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/.github/CONTRIBUTING.md

To be consistent, `_impl` should probably be used everywhere in BLAS1.